gputop/gputop-ncurses.c

```diff
@@ -1623,6 +1630,9 @@ run_ncurses_ui(void *arg)
     }
 #endif
 
+    signal(SIGTSTP, &gputop_terminal_window_exit);
+    signal(SIGINT, &gputop_terminal_window_exit);
```

Comment:
Being a little fussy about what's ok to call inside a signal handler, `perf_tab_leave()` encapsulates quite a bit of code which probably isn't all signal-safe.

There's a nifty system call on Linux called `signalfd()` which is good for this kind of case which lets us get an `fd` that we can poll on via the mainloop we have so we can have a callback which doesn't need to be signal-safe.

It needs a little more boilerplate, but can be used something like:
```c
sigset_t mask;
sigemptyset (&mask);
sigaddset (&mask, SIGTERM);
sigaddset (&mask, SIGINT);

if (sigprocmask(SIG_BLOCK, &mask, NULL) < 0) {
    fprintf(stderr, "failed to block default SIGINT handler: %m");
    return NULL;
}

int fd = signalfd(-1, &mask, SFD_CLOEXEC);

uv_poll_init(gputop_mainloop, &signal_poll, fd);
uv_poll_start(&signal_poll, POLLIN, signal_handler_cb);
```
(would need to move down after we initialize `gputop_mainloop` + would e.g. need a static global `uv_poll_t signal_poll;`)

I don't think we should be responding to SIGSTOP here, just SIGINT and SIGTERM. Ctrl-Z sends a SIGSTOP but that's not supposed to cause a program to exit, it causes it to freeze until a SIGCONT is signaled.